src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -547,6 +573,38 @@ Subscriber* DomainParticipantImpl::create_subscriber(
     return sub;
 }
 
+Topic* DomainParticipantImpl::create_topic(
+        const std::string& topic_name,
+        const std::string& type_name,
+        const TopicQos& qos,
+        TopicListener* listener,
+        const StatusMask& mask)
+{
+    //Look for the correct type registration
+    TypeSupport type_support = find_type(type_name);
+    if (type_support.empty())
+    {
+        logError(PARTICIPANT, "Type : " << type_name << " Not Registered");
+        return nullptr;
+    }
+
+    if (! TopicImpl::check_qos(qos))
+    {
+        logError(PARTICIPANT, "TopicQos inconsistent or not supported");
+        return nullptr;
+    }
+
+    //TODO CONSTRUIR LA IMPLEMENTACION DENTRO DEL OBJETO DEL USUARIO.
+    TopicImpl* topic_impl = new TopicImpl(this, type_support, qos, listener);
+    Topic* topic = new Topic(topic_name, topic_impl, mask);
+
```

Comment:
Create also a new Instance Handle and GUID for the new Topic
```suggestion
    GUID_t topic_guid = guid();
    do
    {
        topic_guid.entityId = fastrtps::rtps::c_EntityId_Unknown;
        rtps_participant_->get_new_entity_id(topic_guid.entityId);
    } while (exists_entity_id(topic_guid.entityId));
    } while (exists_entity_id(sub_guid.entityId));

InstanceHandle_t topic_handle(topic_guid);
    topic_impl->handle_ = topic_handle;
```